hitman3.asl

```diff
@@ -1,58 +1,356 @@
+	*/
+
+    var e_lfanew = memory.ReadValue<int>(baseAddress + 0x3C);
+    var SizeOfOptionalHeader = memory.ReadValue<short>(baseAddress + e_lfanew + 0x4 + 0x10);
+    var textSectionBase = baseAddress + e_lfanew + 0x18 + SizeOfOptionalHeader;
+
+    var textVirtualSize = memory.ReadValue<int>(textSectionBase + 0x08);
+    var textVirtualAddress = memory.ReadValue<int>(textSectionBase + 0x0C);
+
+    var textSectionData = memory.ReadBytes(baseAddress + textVirtualAddress, textVirtualSize);
+
+    //We probably could use internaly class SignatureScanner but it dosen't support multiple signatures at the same time
+    //
+
+    for (var i = 0; i < textSectionData.Length; i++)
+    {
+        bool finish = true;
+
+        foreach (var signature in signaturesList)
+        {
+            if (signature.state[0] == 1)
+                continue;
+
+            finish = false;
+
+            var currentIndex = signature.state[1];
+
+            if (signature.sig[currentIndex] == 0xFF
+            || signature.sig[currentIndex] == textSectionData[i])
+            {
+                if ((currentIndex + 1) == signature.sig.Count)
+                {
+                    var sigAddress = baseAddress + textVirtualAddress + (i - currentIndex);
+
+                    if (signature.type == "offset")
+                    {
+                        signature.state[2] = memory.ReadValue<int>(sigAddress + signature.read);
+                    }
+                    else if (signature.type == "pointer")
+                    {
+                        signature.state[2] = memory.ReadValue<int>(sigAddress + signature.read) + textVirtualAddress + (i - currentIndex) + 0x7;
+                    }
+
+                    signature.state[0] = 1;
+                }
+                signature.state[1]++;
+                continue;
+            }
+
+            signature.state[1] = 0;
+        }
+
+        if (finish)
+            break;
+    }
+
+    var gameAddresses = new Dictionary<string, int>();
+
+    foreach (var signature in signaturesList)
+    {
+        gameAddresses[signature.name] = signature.state[2];
+    }
+
+    vars.timePointer = new DeepPointer(gameAddresses["gameTime"]);
+    vars.hudMissionTimeControllerPointer = new DeepPointer(gameAddresses["hudMissionTimeController"]);
+    vars.missionTimePointer = new DeepPointer(gameAddresses["hudMissionTimeController"], 0x40);
+    vars.gateExitedPointer = new DeepPointer(gameAddresses["contractsManager"] + gameAddresses["gateExited"]);
+    vars.metadataLocationPointer = new DeepPointer(gameAddresses["contractsManager"] + gameAddresses["metadataLocation"], 0x00);
+
+    //I don't know how to modify pointers in state so will create my own variables
     //
-    // Correct offsets is 0x110
-    // F3 0F 11 87 ? ? ? ? F3 0F 10 87 ? ? ? ? 0F 2F 05 ? ? ? ?  3.110.1.0 sig
-    //   movss   dword ptr [rdi+110h], xmm0
+    // Fix me please?
     //
-    float ALT_IGT: 0x03C81960, 0x110;
+
+    vars.currentTime = (ulong)0;
```

Comment:
why ulong and not double?

Reply:
Game has time in that format and when wants to display it converts it to float via formula which i provided in gameTime

Reply:
Gotcha. I guess I'd add a comment that says "game time in nanoseconds -- this is the format the game uses internally for reference time" or something to that effect